cpp/arcticdb/util/trace.hpp

```diff
@@ -19,6 +19,10 @@ namespace arcticdb {
 std::string get_stack();
 std::string get_type_name(const std::type_info & );
 
+/*
+ *  Backwards compatibility layer for old uses of the Python API, for which this was the base exception
+ *  type raised from the extension module.
+ */
 class ArcticNativeCxxException : public std::exception {
```

Comment:
I'm a teeny bit sad that we have to have ArcticNativeCxx anything in ArcticDB, but I'm guessing there was no other way of doing it. 

Can we definitely not catch and rethrow in the internal wrapper?

Reply:
Yeah. We might be able to do it in the wrapper but I actually I have no idea how, since it's really just a thing that re-writes imports. I really like the idea of keeping the wrapper as just a straight pass through because there will be less chance of breaking internal side with an ArcticDB change.